server/controllers/updateMemosCtrl.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const { safeGet } = require('safe-utils')
+const log = require('@kth/log')
+const { getKoppsCourseRoundTerms, getAllCourseCodes, getApplicationFromLadokUID } = require('../koppsApi')
+const { getMemoApiData, changeMemoApiData } = require('../kursPmDataApi')
+
+async function _fetchAllMemosAndUpdateMemoWithApplicationCodes() {
+  const courseCodes = await getAllCourseCodes()
+  let totalMemosToUpdate = 0
+  let totalCoursesFetchedFromKopps = 0
+  let totalApplicationCodesFetchedFromKopps = 0
+  if (courseCodes && courseCodes.length > 0) {
+    totalCoursesFetchedFromKopps = courseCodes.length
+    for await (const courseCode of courseCodes) {
+      const memoData = await getMemoApiData('getAllMemosByCourseCode', {
```

Comment:
It might happen that some of the course/memos or variable extractions break, in this case it will break script and requires to start over. In such cases recommendation is to add try/catch and have a safer extraction of variables